front/lib/api/assistant/actions/tables_query.ts

```diff
@@ -156,291 +155,285 @@ async function tablesQueryActionSpecification({
+      conversation,
+      agentMessage,
+      rawInputs,
+      functionCallId,
+      step,
+    }: BaseActionRunParams
+  ): AsyncGenerator<
+    | TablesQueryErrorEvent
+    | TablesQuerySuccessEvent
+    | TablesQueryParamsEvent
+    | TablesQueryOutputEvent
+  > {
+    const owner = auth.workspace();
+    if (!owner) {
+      throw new Error("Unexpected unauthenticated call to `runQueryTables`");
+    }
 
-  const question = rawInputs.question as string;
+    const { actionConfiguration } = this;
 
-  let output: Record<string, string | boolean | number> = {};
+    if (!rawInputs.question || typeof rawInputs.question !== "string") {
+      yield {
+        type: "tables_query_error",
+        created: Date.now(),
+        configurationId: agentConfiguration.sId,
+        messageId: agentMessage.sId,
+        error: {
+          code: "tables_query_parameters_generation_error",
+          message: `Error generating parameters for tables query: failed to generate a valid question.`,
+        },
+      };
+      return;
+    }
 
-  // Creating action
-  const action = await AgentTablesQueryAction.create({
-    tablesQueryConfigurationId: configuration.sId,
-    params: rawInputs,
-    output,
-    functionCallId,
-    agentMessageId: agentMessage.agentMessageId,
-    step: step,
-  });
+    const question = rawInputs.question as string;
 
-  yield {
-    type: "tables_query_params",
-    created: Date.now(),
-    configurationId: configuration.sId,
-    messageId: agentMessage.sId,
-    action: new TablesQueryAction({
-      id: action.id,
-      params: action.params as DustAppParameters,
-      output: action.output as Record<string, string | number | boolean>,
-      functionCallId: action.functionCallId,
-      functionCallName: action.functionCallName,
-      agentMessageId: action.agentMessageId,
-      step: action.step,
-    }),
-  };
+    let output: Record<string, string | boolean | number> = {};
 
-  // Generating configuration
-  const config = cloneBaseConfig(
-    DustProdActionRegistry["assistant-v2-query-tables"].config
-  );
-  const tables = actionConfiguration.tables.map((t) => ({
-    workspace_id: t.workspaceId,
-    table_id: t.tableId,
-    data_source_id: t.dataSourceId,
-  }));
-  config.DATABASE_SCHEMA = {
-    type: "database_schema",
-    tables,
-  };
-  config.DATABASE = {
-    type: "database",
-    tables,
-  };
-  config.DATABASE_TABLE_HEAD = {
-    type: "database",
-    tables,
-  };
-
-  // Running the app
-  const res = await runActionStreamed(
-    auth,
-    "assistant-v2-query-tables",
-    config,
-    [
-      {
-        question,
-        instructions: configuration.instructions,
-      },
-    ],
-    {
-      conversationId: conversation.sId,
-      workspaceId: conversation.owner.sId,
-      agentMessageId: agentMessage.sId,
-    }
-  );
+    // Creating action
+    const action = await AgentTablesQueryAction.create({
+      tablesQueryConfigurationId: actionConfiguration.sId,
```

Comment:
🚗 We were using the wrong `sId` here.

Reply:
If we don't use it (and I don't think we do), we might actually want to drop the field.
Will investigate 👍

Reply:
Keep me posted :)